[docs] Add pickle security warning to package docs

SplitInfinity opened the pull request:
Stack from [ghstack](https://github.com/ezyang/ghstack):
* **#59959 [docs] Add pickle security warning to package docs**

**Summary**
This commit replaces the warning on the `torch.package` documentation
page about the module not being publicly released (which will no longer
be true as of 1.9) with one that warns about security issues caused by
the use of the `pickle` module.

**Test Plan**
1) Built the docs locally.
2) Continuous integration.

<img width="877" alt="Captura de Pantalla 2021-06-14 a la(s) 11 22 05 a  m" src="https://user-images.githubusercontent.com/4392003/121940300-c98cab00-cd02-11eb-99dc-08e29632079a.png">

Differential Revision: [D29108429](https://our.internmc.facebook.com/intern/diff/D29108429)

Commit:
[docs] Add pickle security warning to package docs

**Summary**
This commit replaces the warning on the `torch.package` documentation
page about the module not being publicly released (which will no longer
be true as of 1.9) with one that warns about security issues caused by
the use of the `pickle` module.

**Test Plan**
1) Built the docs locally.
2) Continuous integration.

[ghstack-poisoned]

suo commented:
Can we add a link to the pickle docs? As well as an explanation of what the security issue is (running untrusted code during unpickling) instead of just saying it's not secure.

suo commented on docs/source/package.rst:
```diff
@@ -5,7 +5,7 @@ torch.package
 .. warning::
 
-    This module is experimental and has not yet been publicly released.
+    This module depends on the ``pickle`` module which is is not secure. Only unpackage data you trust.
```
Let's put the warning after the overview, before the toc

Commit:
Update on "[docs] Add pickle security warning to package docs"


**Summary**
This commit replaces the warning on the `torch.package` documentation
page about the module not being publicly released (which will no longer
be true as of 1.9) with one that warns about security issues caused by
the use of the `pickle` module.

**Test Plan**
1) Built the docs locally.
2) Continuous integration.

<img width="875" alt="Captura de Pantalla 2021-06-14 a la(s) 11 11 07 a  m" src="https://user-images.githubusercontent.com/4392003/121938933-4159d600-cd01-11eb-9133-ac3fe73438c5.png">



[ghstack-poisoned]

Commit:
Update on "[docs] Add pickle security warning to package docs"


**Summary**
This commit replaces the warning on the `torch.package` documentation
page about the module not being publicly released (which will no longer
be true as of 1.9) with one that warns about security issues caused by
the use of the `pickle` module.

**Test Plan**
1) Built the docs locally.
2) Continuous integration.

<img width="877" alt="Captura de Pantalla 2021-06-14 a la(s) 11 22 05 a  m" src="https://user-images.githubusercontent.com/4392003/121940300-c98cab00-cd02-11eb-99dc-08e29632079a.png">



[ghstack-poisoned]

SplitInfinity commented:
@splitinfinity has imported this pull request.  If you are a Facebook employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D29108429).
